ocs_ci/ocs/resources/mcg.py

```diff
@@ -38,21 +37,29 @@ def __init__(self):
         """
         Constructor for the MCG class
         """
-
-        # Todo: find a better solution for not being able to verify requests with a self-signed cert
-        logger.warning('Suppressing InsecureRequestWarnings')
-        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
-
         self.namespace = config.ENV_DATA['cluster_namespace']
         ocp_obj = OCP(kind='noobaa', namespace=self.namespace)
+        self.core_pod = Pod(
+            **get_pods_having_label(constants.NOOBAA_CORE_POD_LABEL, self.namespace)[0]
+        )
+
+        # Copy the self-signed certificate to the local runner
+        # For usage with boto3 (in case the cert isn't found)
+        if not os.path.isfile(constants.MCG_CRT_LOCAL_PATH):
+            ocp_obj.exec_oc_cmd(
+                f'cp {self.core_pod.name}:'
+                f'{constants.MCG_CRT_REMOTE_PATH} '
+                f'{constants.MCG_CRT_LOCAL_PATH}'
+            )
+
         results = ocp_obj.get()
         self.s3_endpoint = (
             results.get('items')[0].get('status').get('services')
-            .get('serviceS3').get('externalDNS')[-1]
+            .get('serviceS3').get('externalDNS')[0]
```

Comment:
Why is this change needed? I am just curious.

Reply:
Great question!

NooBaa actually provides two addresses to each endpoint. For example - 
```
  services:
    serviceMgmt:
      externalDNS:
      - https://noobaa-mgmt-openshift-storage.apps.<cluster-name>.qe.rh-ocs.com
      - https://<UUID>.us-east-2.elb.amazonaws.com:443
```

The first one leads to the NooBaa DNS entry (the one that's also signed by the certificate), while the second leads directly to the AWS load balancer (that isn't covered by the certificate).